lib/repository.js

```diff
@@ -648,14 +649,20 @@ Repository.prototype.fetchAll = function(
   var repo = this;
 
   return repo.getRemotes().then(function(remotes) {
-    var fetchPromises = [];
+    var fetchPromise = Promise.resolve();
 
     remotes.forEach(function(remote) {
-      fetchPromises.push(
-        repo.fetch(remote, remoteCallbacks, autoTag, pruneAfter, callback));
+      fetchPromise.then(function() {
+        repo.fetch(remote, remoteCallbacks, autoTag, pruneAfter, callback);
```

Comment:
Isn't callback here supposed to be resolve? Or something else? I don't get how this could work